src/App.tsx

```diff
@@ -1,14 +1,89 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { ChangeEvent, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos, getUser } from './api';
+import { Todo } from './types/Todo';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [searchQuery, setQuery] = useState('');
+  const [filteredTodos, setFilteredTodos] = useState(todos);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isSearchActive, setSearchActive] = useState(false);
+  const [isLoadingUser, setIsLoadingUser] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const handleInput = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    setSearchActive(true);
+  };
+
+  const selectTodo = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const closePopUp = () => {
+    setSelectedTodo(null);
+  };
+
+  const onClearInput = () => {
+    setQuery('');
+    setSearchActive(false);
+  };
+
+  const visibleTodos = filteredTodos.filter(todo => {
+    const lowerTitle = todo.title.toLowerCase();
+    const lowerQuery = searchQuery.toLowerCase().trim();
+
+    return lowerTitle.includes(lowerQuery);
+  });
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(data => {
+        setTodos(data);
+        setFilteredTodos(data);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, []);
+
+  useEffect(() => {
+    setIsLoadingUser(true);
+    if (selectedTodo) {
+      getUser(selectedTodo.userId)
+        .then(user => {
+          setSelectedUser(user);
+        })
+        .finally(() => {
+          setIsLoadingUser(false);
+        });
+    }
+  }, [selectedTodo]);
+
+  const getSelectTodos = (event: ChangeEvent<HTMLSelectElement>) => {
+    switch (event.target.value) {
+      case 'active':
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+      case 'completed':
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+      default:
+        setFilteredTodos(todos);
+    }
+  };
```

Comment:
it is better to create enum for that and operate with it. String values are really bad options